ci/docker/ubuntu-20.04-cpp.dockerfile

```diff
@@ -129,6 +130,7 @@ ENV ARROW_BUILD_TESTS=ON \
     ARROW_WITH_ZSTD=ON \
     ASAN_SYMBOLIZER_PATH=/usr/lib/llvm-${llvm}/bin/llvm-symbolizer \
     AWSSDK_SOURCE=BUNDLED \
+    GOOGLE_CLOUD_CPP_SOURCE=BUNDLED \
```

Comment:
```suggestion
    google_cloud_cpp_storage_SOURCE=BUNDLED \
```

Reply:
Done. Note that this could be a nuisance if you ever want to use Bigtable or Spanner as data sources.

Reply:
We may need to implement custom `resolve_dependency()` like feature for them when we also want to use other Google Cloud C++ libraries.

If Google Cloud C++ provides `google_cloud_cpp` CMake package with `storage`, `bigtable` and so on components, we can use `google_cloud_cpp_SOURCE` and `find_package(google_cloud_cpp COMPONENTS storage bigtable ...)`.